R/tar-terra-vect.R

```diff
@@ -29,68 +36,144 @@
 #'   })
 #' }
 tar_terra_vect <- function(name,
-                          command,
-                          pattern = NULL,
-                          packages = targets::tar_option_get("packages"),
-                          tidy_eval = targets::tar_option_get("tidy_eval"),
-                          library = targets::tar_option_get("library"),
-                          repository = targets::tar_option_get("repository"),
-                          iteration = targets::tar_option_get("iteration"),
-                          error = targets::tar_option_get("error"),
-                          memory = targets::tar_option_get("memory"),
-                          garbage_collection = targets::tar_option_get("garbage_collection"),
-                          deployment = targets::tar_option_get("deployment"),
-                          priority = targets::tar_option_get("priority"),
-                          resources = targets::tar_option_get("resources"),
-                          storage = targets::tar_option_get("storage"),
-                          retrieval = targets::tar_option_get("retrieval"),
-                          cue = targets::tar_option_get("cue")) {
-  name <- targets::tar_deparse_language(substitute(name))
-
-  envir <- targets::tar_option_get("envir")
-
-  command <- targets::tar_tidy_eval(
-    expr = as.expression(substitute(command)),
-    envir = envir,
-    tidy_eval = tidy_eval
-  )
-  pattern <- targets::tar_tidy_eval(
-    expr = as.expression(substitute(pattern)),
-    envir = envir,
-    tidy_eval = tidy_eval
-  )
-
-  format_terra_shapefile_zip <- targets::tar_format(
-      read = function(path) terra::vect(paste0("/vsizip/{", path, "}")),
-      write = function(object, path) {
-          terra::writeVector(
-              x = object,
-              filename = paste0(path, ".shz"),
-              filetype = "ESRI Shapefile"
-          )
-          file.rename(paste0(path, ".shz"), path)
-      },
-      marshal = function(object) terra::wrap(object),
-      unmarshal = function(object) terra::unwrap(object)
-  )
-
-  targets::tar_target_raw(
-    name = name,
-    command = command,
-    pattern = pattern,
-    packages = packages,
-    library = library,
-    format = format_terra_shapefile_zip,
-    repository = repository,
-    iteration = iteration,
-    error = error,
-    memory = memory,
-    garbage_collection = garbage_collection,
-    deployment = deployment,
-    priority = priority,
-    resources = resources,
-    storage = storage,
-    retrieval = retrieval,
-    cue = cue
-  )
+                           command,
+                           pattern = NULL,
+                           filetype = NULL,
+                           gdal = NULL,
```

Comment:
Is there any reason not to do things like this?
```suggestion
                           filetype = geotargets_option_get("gdal.raster.driver"),
                           gdal = geotargets_option_get("gdal.raster.creation_options"),
```
I think that would simplify the logic of the unexported functions and make it more obvious how users can change the defaults.

Reply:
Opened an issue for this (#34)